File: android/src/main/cpp/common/ExtensionSdkDemoParamsTypeE.cpp
//
// Created by asterisk on 2022/3/1.
//

#include "ExtensionSdkDemoParamsTypeE.h"

E_SDK_NAMESPACE_BEGIN

const std::map<ExtensionSdkDemoParamsTypeEValue, std::string>
list = {
        {ExtensionSdkDemoParamsTypeEValue::PARAMS_JSON,   "json"},
        {ExtensionSdkDemoParamsTypeEValue::PARAMS_Object, "object"},
        {ExtensionSdkDemoParamsTypeEValue::PARAMS_PB,     "proto"},
};

E_SDK_NAMESPACE_END

File: android/src/main/java/com/example/extension_sdk_demo/common/ExtensionSdkDemoParamsTypeE.java
package com.example.extension_sdk_demo.common;

/// 参数类型: 方便以后参数的灵活性和扩展性
public enum ExtensionSdkDemoParamsTypeE {
    PARAMS_JSON("json"),
    PARAMS_Object("object"),
    PARAMS_PB("proto");

    private final String paramsType;

    ExtensionSdkDemoParamsTypeE(String type) {
        this.paramsType = type;
    }

    public String getParamsType() {
        return paramsType;
    }
}
